frontend-plugin-core/src/main/java/com/github/eirslett/maven/plugins/frontend/lib/NodeVersionHelper.java

```diff
@@ -0,0 +1,306 @@
+            "node-v0.4.0",
+            "node-v0.4.1",
+            "node-v0.4.10",
+            "node-v0.4.11",
+            "node-v0.4.12",
+            "node-v0.4.2",
+            "node-v0.4.3",
+            "node-v0.4.4",
+            "node-v0.4.5",
+            "node-v0.4.6",
+            "node-v0.4.7",
+            "node-v0.4.8",
+            "node-v0.4.9",
+            "node-v0.4",
+            "node-v0.5.0",
+            "node-v0.6.1",
+            "node-v0.6.10",
+            "node-v0.6.11",
+            "node-v0.6.12",
+            "node-v0.6.13",
+            "node-v0.6.2",
+            "node-v0.6.3",
+            "node-v0.6.4",
+            "node-v0.6.5",
+            "node-v0.6.6",
+            "node-v0.6.7",
+            "node-v0.6.8",
+            "node-v0.6.9"
+    ).collect(toSet());
+
+    @VisibleForTesting
+    static final Pattern VALID_VERSION_PATTERN = Pattern.compile("^v?\\d*\\.\\d*\\.\\d*$");
+
+    private static final String LATEST_OF_GIVEN_MAJOR_FORMAT = "latest-v%s.x";
+    private static final Pattern MAJOR_VERSION_ONLY_PATTERN = Pattern.compile("^\\d+$");
+
+    /**
+     * We cache this to prevent multiple network requests in a multi-module Maven build, it's very unlikely a version
+     * someone is released suddenly halfway through the build and that a consumer has actually asked for that
+     */
+    @VisibleForTesting
+    static final AtomicReference<Optional<List<NodeVersion>>> nodeVersions = new AtomicReference<>(empty());
+
+    public static boolean validateVersion(String version) {
+        if (UNUSUAL_VALID_VERSIONS.contains(version)) {
+            return true;
+        }
+
+        version = version.replaceFirst("v", ""); // we're about to add it back
+        if (MAJOR_VERSION_ONLY_PATTERN.matcher(version).find()) {
+            String latestOfMajorVersion = format(LATEST_OF_GIVEN_MAJOR_FORMAT, version).toLowerCase();
+            return UNUSUAL_VALID_VERSIONS.contains(latestOfMajorVersion);
+        }
+
+        Matcher matcher = VALID_VERSION_PATTERN.matcher(version);
+        return matcher.find();
+    }
+
+    public static String getDownloadableVersion(String version) {
+        version = version.toLowerCase(); // all the versions seem to be lower case
+
+        if (UNUSUAL_VALID_VERSIONS.contains(version)) {
+            return version;
+        }
+
+        version = version.replaceFirst("v", ""); // we're about to add it back
+
+        return findMatchingReleasedVersion(version)
+                .orElse("v" + version);
+    }
+
+    public static Optional<String> findMatchingReleasedVersion(String requestedVersionLowercaseWithoutLeadingV) {
+        if (!nodeVersions.get().isPresent()) {
+            synchronized (NodeVersionHelper.class) { // avoiding racing sending multiple requests
+                try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                    HttpGet getNodeVersionIndex = new HttpGet("https://nodejs.org/dist/index.json");
```

Comment:
Should this be based on `nodeDownloadRoot` property maybe so that we can override node registry?